ext/opcache/Optimizer/sccp.c

```diff
@@ -844,13 +844,20 @@ static inline int ct_eval_func_call(
 				|| zend_string_equals_literal(name, "urldecode")
 				|| zend_string_equals_literal(name, "rawurlencode")
 				|| zend_string_equals_literal(name, "rawurldecode")
-				|| zend_string_equals_literal(name, "php_uname")) {
+				|| zend_string_equals_literal(name, "php_uname")
+				|| zend_string_equals_literal(name, "dirname")
+				|| zend_string_equals_literal(name, "basename")
+				|| zend_string_equals_literal(name, "md5")
+				|| zend_string_equals_literal(name, "crc32")
+				|| zend_string_equals_literal(name, "sha1")) {
 			if (Z_TYPE_P(args[0]) != IS_STRING) {
 				return FAILURE;
 			}
 			/* pass */
 		} else if (zend_string_equals_literal(name, "array_keys")
-				|| zend_string_equals_literal(name, "array_values")) {
+				|| zend_string_equals_literal(name, "array_values")
+				|| zend_string_equals_literal(name, "array_unique")
```

Comment:
I would have guessed that there was a requirement that the types in questions don't emit notices? Here, PHP can emit notices if the values cannot be converted to strings (and the configured error handler can throw, should not happen when the line of code is compiled, not executed)

```
php > var_export(array_unique([[], []]));
Notice: Array to string conversion in php shell code on line 1
Notice: Array to string conversion in php shell code on line 1
array (
  0 => 
  array (
  ),
)
```

It's solvable with checking that all elements of `args[0]` are known to be able to cast to string without emitting notices (see other uses of ZEND_HASH_FOREACH_VAL/ZEND_HASH_FOREACH_END)

(i.e. null, bool, int, string, float)